src/logicLayer/BattleForecast.java

```diff
@@ -0,0 +1,75 @@
+package logicLayer;
+
+public class BattleForecast {
+	private Unit userUnit;
+	private Unit enemyUnit;
+	
+	//This will contain attack amounts, hit percentages, and crit percentages
+	private int[] forecastData = new int[6];
+	
+	public BattleForecast(Unit aUserUnit, Unit aEnemyUnit) {
+		this.userUnit = aUserUnit;
+		this.enemyUnit = aEnemyUnit;
+	}
+	
+	public Unit getUserUnit() {
+		return new Unit(this.userUnit);
+	}
+	public Unit getEnemyUnit() {
+		return new Unit(this.enemyUnit);
+	}
+	
+	
+	public int[] getForecastData() {
+		Tile userTile = this.userUnit.getTile();
+		Tile enemyTile = this.enemyUnit.getTile();
+		
+		Unit userUnit = userTile.getUnit();
+		Unit enemyUnit = enemyTile.getUnit();
+		
+		if (userTile.hasUnit() &&
+			enemyTile.hasUnit()) {
+			int attForecastDamage = calculateDamage(userUnit, enemyUnit);
+			int defForecastDamage = calculateDamage(enemyUnit, userUnit);
+			
+			int attForecastPercent = calculateHitPercent(userUnit, enemyUnit);
+			int defForecastPercent = calculateHitPercent(enemyUnit, userUnit);
+			
+			int attCritPercent = calculateCritPercent(userUnit, enemyUnit);
+			int defCritPercent = calculateCritPercent(userUnit, enemyUnit);
+			
+			forecastData[0] = attForecastDamage;
+			forecastData[1] = defForecastDamage;
+			forecastData[2] = attForecastPercent;
+			forecastData[3] = defForecastPercent;
+			forecastData[4] = attCritPercent;
+			forecastData[5] = defCritPercent;
+		}
+		return forecastData;
+	}
+	
+	private int calculateDamage(Unit attUnit, Unit defUnit) {
+		int oneHit = attUnit.getStats().getAtt().getCurrentValue() - defUnit.getStats().getDef().getCurrentValue();
+		return oneHit;
+	}
+	
+	private int calculateHitPercent(Unit attUnit, Unit defUnit) {
+		double GLOBALHITMULTIPLIER = 1.2;

```

Comment:
constants are usually declared at the top like so:
private static final double GLOBALHITMULTIPLIER = 1.2;
*note that the "static" isn;t necessary, but nice to have